.github/workflows/tests.yml

```diff
@@ -0,0 +1,47 @@
+name: tests
+
+permissions:
+    id-token: write  # required for requesting the JWT
+    contents: read  # required for actions/checkout
+
+on:
+    push:
+        branches:
+            - "tests-infra-pytest"
+
+jobs:
+    python-job:
+        name: "PyTest tests"
+        runs-on: ubuntu-latest
+        environment: "test"
+
+        steps:
+          - name: Checkout repository
+            uses: actions/checkout@v3
+
+          - name: Setup Python
+            uses: actions/setup-python@v3
+            with:
+                python-version: '3.11.4'
+
+          - name: Assume Github OIDC role
+            uses: aws-actions/configure-aws-credentials@v2
+            with:
+                aws-region: us-west-2
+                role-to-assume: arn:aws:iam::916098889494:role/GithubOIDCRole-MAAP-Project-maap-eoapi-test
+                role-session-name: maap-eoapi-test
+
+          - name: Install dependencies
+            run: |
+                python -m pip install --upgrade pip
+                pip install -r tests/requirements.txt
+
+          - name: Run pytest
+            env:
+                INGESTOR_URL: ${{ vars.INGESTOR_URL }}
```

Comment:
Note for later. Right now, the stac ingestor URL isn't something we choose, it's automatically determined by AWS. If it gets changed because we change something in the ingestor and re-deploy, these tests will fail until we update the ingestor URL in the environment. 

Two solutions to this. 

- we create a record for the ingestor using the `maap-project.org` domain, something like `stac-ingestor.test.maap-project.org` (for the test one)
- we write a script that programmatically finds the current URL and have the test github action run that script (instead of pulling the URL from a variable).